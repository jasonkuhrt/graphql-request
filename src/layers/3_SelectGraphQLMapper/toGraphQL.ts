import type { Select } from '../2_Select/__.js'
import type { SchemaDrivenDataMap } from '../7_customScalars/generator/SchemaDrivenDataMap.js'
import { toGraphQLDocument } from './nodes/Document.js'

export interface Options {
  sddm: SchemaDrivenDataMap
  extractOperationVariables?: boolean
}

export const toGraphQL = (input: {
  document: Select.Document.DocumentNormalized
  options?: Options
}) => {
  return toGraphQLDocument(input.document, input.options)
}
