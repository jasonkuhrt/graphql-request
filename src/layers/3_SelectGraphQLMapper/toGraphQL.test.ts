import { describe, expect, test } from 'vitest'
import { db } from '../../../tests/_/schemas/db.js'
import { $index } from '../../../tests/_/schemas/kitchen-sink/graffle/modules/RuntimeCustomScalars.js'
import type * as SelectionSets from '../../../tests/_/schemas/kitchen-sink/graffle/modules/SelectionSets.js'
import { Grafaid } from '../../lib/grafaid/__.js'
import { Select } from '../2_Select/__.js'
import { toGraphQL } from './toGraphQL.js'

type CasesQuery = [selectionSet: SelectionSets.Query]
const testEachQuery = test.for.bind(test)<CasesQuery>

type CasesDescriptiveQuery = [description: string, selectionSet: SelectionSets.Query]
const testEachQueryWithDescription = test.for.bind(test)<CasesDescriptiveQuery>

const tester = [
  `Query - %s`,
  (args: CasesQuery | CasesDescriptiveQuery) => {
    const [description, graffleQuery] = args.length === 1 ? [undefined, args[0]] : args

    const { document, operationsVariables } = toGraphQL({
      document: Select.Document.createDocumentNormalizedFromQuerySelection(graffleQuery as any),
      options: {
        sddm: $index,
        extractOperationVariables: true,
      },
    })

    const beforeAndAfter = `\n`
      + `\n--------------GRAFFLE QUERY-------------\n`
      + JSON.stringify(graffleQuery, null, 2)
      + `\n--------GRAPHQL DOCUMENT & VARIABLES--------\n`
      + Grafaid.Nodes.print(document)
      + `\n----------------\n`
      + JSON.stringify(operationsVariables, null, 2)
      + `\n`

    expect(beforeAndAfter).toMatchSnapshot(description)
  },
] as const

describe(`inline fragment`, () => {
  describe(`field group`, () => {
    testEachQueryWithDescription([
      [`one`, { ___: { __typename: true } }],
      [`multiple`, { ___: [{ __typename: true }, { abcEnum: true }] }],
      [`interface`, { interface: { ___: { __typename: true } } }],
      [`union`, { unionFooBar: { ___: { __typename: true } } }],
      [`directive`, { ___: { __typename: true, $include: true } }],
    ])(...tester)
  })

  describe(`type condition union`, () => {
    testEachQueryWithDescription([
      [`__typename (no fragment)`, { unionFooBar: { __typename: true } }],
      [`scalar`, { unionFooBar: { ___on_Bar: { int: true } } }],
      [`directive`, { unionFooBar: { ___on_Bar: { $skip: true, int: true } } }],
      // s({ unionFooBar: { __on_Bar: {} } }), // todo should be static type error
    ])(...tester)
  })
})

describe(`alias`, () => {
  testEachQuery([
    [{ id: [`x`, true] }],
    [{ id: [[`x`, true], [`id2`, true]] }],
    [{ id: [`x`, { $skip: true }] }],
    [{ object: [`x`, { $skip: true, id: true }] }],
  ])(...tester)
})

describe(`arguments`, () => {
  testEachQuery([
    [{ result: { $: { $case: `Object1` }, __typename: true } }],
    [{ stringWithArgs: { $: { boolean: true, float: 1 } } }],
    [{ stringWithArgs: { $: {} } }],
    // s({ objectWithArgs: { $: { id: `` } } }), // todo should be static error
    // s({ objectWithArgs: { $: {} } }), // todo should be static error
    [{ objectWithArgs: { $: { id: `` }, id: true } }],
    [{ objectWithArgs: { $: {}, id: true } }],
  ])(...tester)

  // dprint-ignore
  describe(`custom scalars`, () => {
    testEachQueryWithDescription([
      [`arg field`, { dateArg: { $: { date: db.date0 } } }],
      [`arg field in non-null`, { dateArgNonNull: { $: { date: db.date0 } } }], 
      [`arg field in list`, { dateArgList: { $: { date: [db.date0, new Date(1)] } } }], 
      [`arg field in list (null)`, { dateArgList: { $: { date: null } } }],
      [`arg field in non-null list (with list)`, { dateArgNonNullList: { $: { date: [db.date0, new Date(1)] } } }],
      [`arg field in non-null list (with null)`, { dateArgNonNullList: { $: { date: [null, db.date0] } } }],
      [`arg field in non-null list non-null`, { dateArgNonNullListNonNull: { $: { date: [db.date0, new Date(1)] } } }],
      [`input object field`, { dateArgInputObject: { $: { input: { idRequired: ``, dateRequired: db.date0, date: new Date(1) } } } }],
      [`nested input object field`, { InputObjectNested: { $: { input: { InputObject: { idRequired: ``, dateRequired: db.date0, date: new Date(1) } } } } }]
    ])(...tester)
  })
})

describe(`directive`, () => {
  describe(`$include`, () => {
    testEachQuery([
      [{ object: { $include: true, id: true } }],
      [{ object: { $include: false, id: true } }],
      [{ object: { $include: undefined, id: true } }],
      [{ object: { $include: { if: true }, id: true } }],
      [{ object: { $include: { if: false }, id: true } }],
      [{ object: { $include: { if: undefined }, id: true } }],
      [{ object: { $include: {}, id: true } }],
    ])(...tester)
  })

  describe(`$skip`, () => {
    testEachQuery([
      [{ object: { $skip: true, id: true } }],
      [{ object: { $skip: false, id: true } }],
      [{ object: { $skip: undefined, id: true } }],
      [{ object: { $skip: { if: true }, id: true } }],
      [{ object: { $skip: { if: false }, id: true } }],
      [{ object: { $skip: { if: undefined }, id: true } }],
      [{ object: { $skip: {}, id: true } }],
    ])(...tester)
  })
})

describe(`other`, () => {
  testEachQuery([
    [{ __typename: true }],
    [{ string: true }],
    // [{ string: 1 }],
    // s({ string: false }), // todo should be static error
    [{ id: true, string: false }],
    // [{ id: true, string: 0 }],
    [{ id: true, string: undefined }],
    [{ object: { id: true } }],
    [{ objectNested: { object: { string: true, id: true, int: false } } }],
    [{ objectNested: { object: { string: true, id: true, int: { $skip: true } } } }],
  ])(...tester)
})
