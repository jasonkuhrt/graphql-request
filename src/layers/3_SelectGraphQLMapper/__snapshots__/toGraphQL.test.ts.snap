// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`alias > Query - { id: [ 'x', { 'undefined': true } ] } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "id": [
    "x",
    {
      "$skip": true
    }
  ]
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  x: id @skip(if: true)
}
----------------
{
  "$default": {}
}
"
`;

exports[`alias > Query - { id: [ 'x', true ] } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "id": [
    "x",
    true
  ]
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  x: id
}
----------------
{
  "$default": {}
}
"
`;

exports[`alias > Query - { id: [ [ 'x', true ], [ 'id2', true ] ] } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "id": [
    [
      "x",
      true
    ],
    [
      "id2",
      true
    ]
  ]
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  x: id
  id2: id
}
----------------
{
  "$default": {}
}
"
`;

exports[`alias > Query - { object: [ 'x', { 'undefined': true, id: true } ] } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": [
    "x",
    {
      "$skip": true,
      "id": true
    }
  ]
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  x: object @skip(if: true) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`arguments > Query - { objectWithArgs: { '$': { id: '' }, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "objectWithArgs": {
    "$": {
      "id": ""
    },
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
query ($id: ID) {
  objectWithArgs(id: $id) {
    id
  }
}
----------------
{
  "$default": {
    "id": ""
  }
}
"
`;

exports[`arguments > Query - { objectWithArgs: { '$': {}, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "objectWithArgs": {
    "$": {},
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  objectWithArgs {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`arguments > Query - { result: { '$': { 'undefined': 'Object1' }, __typename: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "result": {
    "$": {
      "$case": "Object1"
    },
    "__typename": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  result(case: Object1) {
    __typename
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`arguments > Query - { stringWithArgs: { '$': { boolean: true, float: 1 } } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "stringWithArgs": {
    "$": {
      "boolean": true,
      "float": 1
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
query ($boolean: Boolean, $float: Float) {
  stringWithArgs(boolean: $boolean, float: $float)
}
----------------
{
  "$default": {
    "boolean": true,
    "float": 1
  }
}
"
`;

exports[`arguments > Query - { stringWithArgs: { '$': {} } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "stringWithArgs": {
    "$": {}
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  stringWithArgs
}
----------------
{
  "$default": {}
}
"
`;

exports[`arguments > custom scalars > Query - arg field > arg field 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "dateArg": {
    "$": {
      "date": "1970-01-01T00:00:00.000Z"
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
query ($date: Date) {
  dateArg(date: $date)
}
----------------
{
  "$default": {
    "date": "1970-01-01T00:00:00.000Z"
  }
}
"
`;

exports[`arguments > custom scalars > Query - arg field in list (null) > arg field in list (null) 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "dateArgList": {
    "$": {
      "date": null
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
query ($date: [Date!]) {
  dateArgList(date: $date)
}
----------------
{
  "$default": {
    "date": null
  }
}
"
`;

exports[`arguments > custom scalars > Query - arg field in list > arg field in list 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "dateArgList": {
    "$": {
      "date": [
        "1970-01-01T00:00:00.000Z",
        "1970-01-01T00:00:00.001Z"
      ]
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
query ($date: [Date!]) {
  dateArgList(date: $date)
}
----------------
{
  "$default": {
    "date": [
      "1970-01-01T00:00:00.000Z",
      "1970-01-01T00:00:00.001Z"
    ]
  }
}
"
`;

exports[`arguments > custom scalars > Query - arg field in non-null > arg field in non-null 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "dateArgNonNull": {
    "$": {
      "date": "1970-01-01T00:00:00.000Z"
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
query ($date: Date!) {
  dateArgNonNull(date: $date)
}
----------------
{
  "$default": {
    "date": "1970-01-01T00:00:00.000Z"
  }
}
"
`;

exports[`arguments > custom scalars > Query - arg field in non-null list (with list) > arg field in non-null list (with list) 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "dateArgNonNullList": {
    "$": {
      "date": [
        "1970-01-01T00:00:00.000Z",
        "1970-01-01T00:00:00.001Z"
      ]
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
query ($date: [Date]!) {
  dateArgNonNullList(date: $date)
}
----------------
{
  "$default": {
    "date": [
      "1970-01-01T00:00:00.000Z",
      "1970-01-01T00:00:00.001Z"
    ]
  }
}
"
`;

exports[`arguments > custom scalars > Query - arg field in non-null list (with null) > arg field in non-null list (with null) 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "dateArgNonNullList": {
    "$": {
      "date": [
        null,
        "1970-01-01T00:00:00.000Z"
      ]
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
query ($date: [Date]!) {
  dateArgNonNullList(date: $date)
}
----------------
{
  "$default": {
    "date": [
      null,
      "1970-01-01T00:00:00.000Z"
    ]
  }
}
"
`;

exports[`arguments > custom scalars > Query - arg field in non-null list non-null > arg field in non-null list non-null 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "dateArgNonNullListNonNull": {
    "$": {
      "date": [
        "1970-01-01T00:00:00.000Z",
        "1970-01-01T00:00:00.001Z"
      ]
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
query ($date: [Date!]!) {
  dateArgNonNullListNonNull(date: $date)
}
----------------
{
  "$default": {
    "date": [
      "1970-01-01T00:00:00.000Z",
      "1970-01-01T00:00:00.001Z"
    ]
  }
}
"
`;

exports[`arguments > custom scalars > Query - input object field > input object field 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "dateArgInputObject": {
    "$": {
      "input": {
        "idRequired": "",
        "dateRequired": "1970-01-01T00:00:00.000Z",
        "date": "1970-01-01T00:00:00.001Z"
      }
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  dateArgInputObject(input: {idRequired: "", dateRequired: {}, date: {}})
}
----------------
{
  "$default": {}
}
"
`;

exports[`arguments > custom scalars > Query - nested input object field > nested input object field 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "InputObjectNested": {
    "$": {
      "input": {
        "InputObject": {
          "idRequired": "",
          "dateRequired": "1970-01-01T00:00:00.000Z",
          "date": "1970-01-01T00:00:00.001Z"
        }
      }
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  InputObjectNested(
    input: {InputObject: {idRequired: "", dateRequired: {}, date: {}}}
  )
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': { if: false }, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$include": {
      "if": false
    },
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @include(if: false) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': { if: true }, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$include": {
      "if": true
    },
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @include(if: true) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': { if: undefined }, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$include": {},
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @include(if: true) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': {}, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$include": {},
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @include(if: true) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': false, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$include": false,
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @include(if: false) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': true, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$include": true,
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @include(if: true) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $include > Query - { object: { 'undefined': undefined, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': { if: false }, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$skip": {
      "if": false
    },
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @skip(if: false) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': { if: true }, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$skip": {
      "if": true
    },
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @skip(if: true) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': { if: undefined }, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$skip": {},
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @skip(if: true) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': {}, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$skip": {},
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @skip(if: true) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': false, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$skip": false,
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @skip(if: false) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': true, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "$skip": true,
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object @skip(if: true) {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`directive > $skip > Query - { object: { 'undefined': undefined, id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`inline fragment > field group > Query - directive > directive 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "___": {
    "__typename": true,
    "$include": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  ... @include(if: true) {
    __typename
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`inline fragment > field group > Query - interface > interface 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "interface": {
    "___": {
      "__typename": true
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  interface {
    ... {
      __typename
    }
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`inline fragment > field group > Query - multiple > multiple 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "___": [
    {
      "__typename": true
    },
    {
      "abcEnum": true
    }
  ]
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  ... {
    __typename
  }
  ... {
    abcEnum
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`inline fragment > field group > Query - one > one 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "___": {
    "__typename": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  ... {
    __typename
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`inline fragment > field group > Query - union > union 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "unionFooBar": {
    "___": {
      "__typename": true
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  unionFooBar {
    ... {
      __typename
    }
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`inline fragment > type condition union > Query - __typename (no fragment) > __typename (no fragment) 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "unionFooBar": {
    "__typename": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  unionFooBar {
    __typename
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`inline fragment > type condition union > Query - directive > directive 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "unionFooBar": {
    "___on_Bar": {
      "$skip": true,
      "int": true
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  unionFooBar {
    ... on Bar @skip(if: true) {
      int
    }
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`inline fragment > type condition union > Query - scalar > scalar 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "unionFooBar": {
    "___on_Bar": {
      "int": true
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  unionFooBar {
    ... on Bar {
      int
    }
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`other > Query - { __typename: true } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "__typename": true
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  __typename
}
----------------
{
  "$default": {}
}
"
`;

exports[`other > Query - { id: true, string: false } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "id": true,
  "string": false
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  id
}
----------------
{
  "$default": {}
}
"
`;

exports[`other > Query - { id: true, string: undefined } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "id": true
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  id
}
----------------
{
  "$default": {}
}
"
`;

exports[`other > Query - { object: { id: true } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "object": {
    "id": true
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  object {
    id
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`other > Query - { objectNested: { object: { string: true, id: true, int: { 'undefined': true } } } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "objectNested": {
    "object": {
      "string": true,
      "id": true,
      "int": {
        "$skip": true
      }
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  objectNested {
    object {
      string
      id
      int @skip(if: true)
    }
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`other > Query - { objectNested: { object: { string: true, id: true, int: false } } } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "objectNested": {
    "object": {
      "string": true,
      "id": true,
      "int": false
    }
  }
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  objectNested {
    object {
      string
      id
    }
  }
}
----------------
{
  "$default": {}
}
"
`;

exports[`other > Query - { string: true } 1`] = `
"

--------------GRAFFLE QUERY-------------
{
  "string": true
}
--------GRAPHQL DOCUMENT & VARIABLES--------
{
  string
}
----------------
{
  "$default": {}
}
"
`;
